python/pyspark/sql/tests/test_functions.py

```diff
@@ -157,6 +159,32 @@ def test_inverse_trig_functions(self):
             for c in cols:
                 self.assertIn(f"{alias}(a)", repr(f(c)))
 
+    def test_reciprocal_trig_functions(self):
+        # SPARK-36683: Add new built-in SQL functions: SEC and CSC
+        list = [0.0, math.pi/6, math.pi/4, math.pi/3, math.pi/2,
+                math.pi, 3 * math.pi / 2, 2 * math.pi]
+
+        df = self.spark.createDataFrame(list, types.DoubleType())
+
+        def to_reciprocal_trig(func):
+            return [1.0 / func(i) if func(i) != 0 else math.inf for i in list]
+
+        def get_values(l):
+            return [j[0] for j in l]
+
+        def assert_close(a, b):
+            c = get_values(b)
+            diff = [abs(v - c[k]) < 1e-6 if v != math.inf else v == c[k]
```

Comment:
If we decide to move `assert_close` to `testing/sqlutils.py`, it's better to consider the case of `-math.inf`.